orttraining/orttraining/core/agent/training_agent.h

```diff
@@ -24,18 +22,11 @@ class TrainingAgent {
     explicit TrainingAgent(InferenceSession* session);
     virtual ~TrainingAgent();
     // For ORTModule.forward()
-    virtual common::Status RunForward(const RunOptions& run_options, onnxruntime::IOBinding& io_binding,
-                                               std::vector<OrtValue>& user_outputs,
-                                               int64_t& run_id) ORT_MUST_USE_RESULT;
+    virtual common::Status RunForward(const RunOptions& run_options, onnxruntime::IOBinding& io_binding, int64_t& run_id) ORT_MUST_USE_RESULT;
     // For ORTModule.backward()
-    common::Status RunBackward(int64_t run_id, const std::vector<OrtValue>& backward_output_grads) ORT_MUST_USE_RESULT;
+    common::Status RunBackward(const RunOptions& run_options, onnxruntime::IOBinding& io_binding, int64_t run_id) ORT_MUST_USE_RESULT;
```

Comment:
>const RunOptions& run_options [](start = 31, length = 29)

Shall we allow user to modify run_options between partial runs? 

Currently in ortmodule, we cache the run_options in ctx and share the same run_option between forward() and backward()

Reply:
You can change run options between partial graph runs, it's not something we retain after a partial run is over. I assume you are suggesting we remove the const in training_agent code?

Reply:
We aren't using anything from RunOptions, so going back to what you had earlier said, i.e too much flexiblity can have unintended consequences, lets leave RunOptions out :) 

---
In reply to: [600909443](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r600909443) [](ancestors = 600909443)